python/fusion_engine_client/applications/p1_capture.py

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env python3
+
+from collections import defaultdict
+from datetime import datetime
+import math
+import os
+import re
+import select
+import socket
+import sys
+import time
+
+import colorama
+
+try:
+    # pySerial is optional.
+    import serial
+    serial_supported = True
+except ImportError:
+    serial_supported = False
+    # Dummy stand-in if pySerial is not installed.
+    class serial:
+        class SerialException: pass
+
+if __package__ is None or __package__ == "":
+    from import_utils import enable_relative_imports
+    __package__ = enable_relative_imports(__name__, __file__)
+
+from .p1_print import \
+    DeviceSummary, add_print_format_argument, print_message, print_summary_table
+from ..parsers import FusionEngineDecoder
+from ..utils import trace as logging
+from ..utils.argument_parser import ArgumentParser, ExtendedBooleanAction
+
+_logger = logging.getLogger('point_one.fusion_engine.applications.p1_capture')
+
+
+def create_transport(descriptor: str):
```

Comment:
I think this file would benefit from type checks, since it's not very obvious what set of functions are supported on the returned transport. So returning a Union here combined with the isinstance checks should catch any mistakes.